.travis.yml

```diff
@@ -2,7 +2,23 @@ language: java
 sudo: false
 jdk:
 - oraclejdk8
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - rm -fr $HOME/.gradle/caches/modules-2/files-*/com.jetbrains.intellij.idea/
+  - rm -fr $HOME/.gradle/caches/modules-2/metadata-*/descriptors/com.jetbrains.intellij.idea/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - $HOME/.m2
 before_install:
 - export MAVEN_OPTS="-Xmx768M"
+branches:
+  only:
+  - master
+  - release
+  - develop
 script:
-- travis_wait 80 ./BuildDevint -q -e eclipse
+- 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then travis_wait 40 ./BuildDevint -q -e eclipse; fi'
```

Comment:
Can `if  ; then ... else ...  fi` with the following one?

Reply:
I myself agree with you. But I don't know whether there any trick inside, so I just follow the [official docs](https://docs.travis-ci.com/user/pull-requests) 

I can have a try.

Reply:
Oh, now I know why they write so in the docs. It's more clear than a single line script. So I tend not to change it now.